Move logic out of CMakeLists.txt

ezzieyguywuf opened the pull request:
Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [x] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- X (not applicable) Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

I am still compiling the code - I'll update this post once I've run all tests successfully.

luzpaz commented on cMake/FreeCAD_Helpers/FindSalomeMESH.cmake:
```diff
@@ -0,0 +1,126 @@
+    # Salome SMESH sources are under src/3rdParty now
+    if(BUILD_SMESH)
+        # set the internal smesh version:
+        # see src/3rdParty/salomonemesh/CMakeLists.txt and commit https://github.com/FreeCAD/FreeCAD/commit/666a3e5 and https://forum.freecadweb.org/viewtopic.php?f=10&t=30838
```
s/salomonemsh/salomesmesh/

ezzieyguywuf replied:
Good catch, I made this change locally but failed to add it to the git repo.

ezzieyguywuf replied:
The "salomonemsh" error is [in master](https://github.com/FreeCAD/FreeCAD/blob/master/CMakeLists.txt#L680). For this initial PR, I wanted to literally copy/paste what is currently in master and simply move it to the cMake sub-directory. In a subsequent future PR I will start going through the actual code/logic and try to clean things like this up.

luzpaz commented on cMake/FreeCAD_Helpers/FindSalomeMESH.cmake:
```diff
@@ -0,0 +1,126 @@
+macro(FindSalomeMESH)
```
Should be `FindSalomeSMESH` (also the file name should be renamed)

ezzieyguywuf replied:
Good catch, I made this change locally but failed to add it to the git repo.

luzpaz commented on cMake/FreeCAD_Helpers/FindSalomeMESH.cmake:
```diff
@@ -0,0 +1,126 @@
+        configure_file(SMESH_Version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/SMESH_Version.h)
+    endif(BUILD_SMESH)
+
+endmacro(FindSalomeMESH)
```
s/FindSalomeMESH/FindSalomeSMESH/

luzpaz commented on cMake/FreeCAD_Helpers/README.md:
```diff
@@ -0,0 +1,19 @@
+This folder will contain an individual cmake file for each FreeCAD "helper".
+
+A "helper" should be a macro or functio that tries (as much as possible) to adhere to the
```
s/functio/function/

luzpaz commented on cMake/FreeCAD_Helpers/README.md:
```diff
@@ -0,0 +1,19 @@
+The idea here is to break up the cmake build system into smaller, more manageable chunks.
+This should make maintenance easier, and should also make troubleshooting a bit less
+painful. Finally, it should also clean up our top-level CMakeLists.txt file a bit, making
+it bit easier for new developpers to jump in and see what's what.
```
s/developpers/developers/

luzpaz commented on cMake/FreeCAD_Helpers/FindFreeType.cmake:
```diff
@@ -0,0 +1,13 @@
+        find_package(Freetype)
+        if(NOT FREETYPE_FOUND)
+            message("===============================================================\n"
+                    "FreeType2 not found. Part module will lack of makeWireString().\n"
```
`"FreeType2 not found. Part module will lack functionality of makeWireString().\n"`

ezzieyguywuf replied:
Again, this error is currently [in master ](https://github.com/FreeCAD/FreeCAD/blob/master/CMakeLists.txt#L963). I would like to leave it as-is for this PR.

ezzieyguywuf commented:
I can confirm that all unit tests now pass on my local system.

wwmayer commented:
https://forum.freecadweb.org/viewtopic.php?f=17&t=39067

wwmayer commented:
If you look on the travis builds then you will realize that it doesn't work on Windows. It cannot find the Qt5 modules.

ezzieyguywuf commented:
@wwmayer I'm aware of the travis issues on Windows. I am working on setting up a build environment in Windows in order to try to duplicate. In the meantime, I've merged in the latest master.

ezzieyguywuf commented:
@wwmayer @luzpaz it seems that this PR is now passing the travis builds. I have squashed the commits back down to one and rebased over the latest master.

wwmayer commented:
Merged.

luzpaz commented:
Quick request, could we add a comment in the top level CMakeList.txt about we've moved most CMake 'modules' in to `./cMake/FreeCAD_Helpers/` ? or is it superfluous ?

ezzieyguywuf commented:
I can add a comment. I was going to prepare some new PRs with cmake updates
/ fixes I’ll put it in one of those

El El jue, sep. 26, 2019 a las 8:38 p. m., luzpaz <notifications@github.com>
escribió:

> Quick request, could we add a comment in the top level CMakeList.txt about
> we've moved most CMake 'modules' in to ./cMake/FreeCAD_Helpers/ ? or is
> it superfluous ?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/FreeCAD/FreeCAD/pull/2477?email_source=notifications&email_token=AAEMB5FL62E5KYOAKTN2OOTQLVI2FA5CNFSM4ISZGESKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7XKZQA#issuecomment-535735488>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEMB5GEH5ZTGAWBYTANPB3QLVI2FANCNFSM4ISZGESA>
> .
>
